legacy-commands-v1/src/main/java/io/github/minecraftcursedlegacy/api/command/CursedCommandSource.java

```diff
@@ -0,0 +1,23 @@
+package io.github.minecraftcursedlegacy.api.command;
+
+import io.github.minecraftcursedlegacy.impl.command.MultiplayerCursedCommandSourceImpl;
+import io.github.minecraftcursedlegacy.impl.command.SingleplayerCursedCommandSourceImpl;
+import net.minecraft.entity.player.Player;
+import net.minecraft.server.command.CommandSource;
+
+import javax.annotation.Nullable;
+
+public interface CursedCommandSource {
+    @Nullable Player getPlayer();
+    String getName();
+    void sendCommandFeedback(String message);
+    void sendError(String message);
+
+    static CursedCommandSource singleplayer(Player player) {
+        return new SingleplayerCursedCommandSourceImpl(player);
+    }
+
+    static CursedCommandSource multiplayer(CommandSource source) {
```

Comment:
won't this crash in client side in prod because of not finding CommandSource when classloading?

Reply:
It hasn't in my testing, I assume because it's never called. I can throw it in a different class if you'd like (I probably should)